upstairs/src/lib.rs

```diff
@@ -498,18 +498,53 @@ async fn process_message(
             )
         }
         Message::ErrorReport {
-            upstairs_id: _,
-            session_id: _,
+            upstairs_id,
+            session_id,
             job_id,
             error,
         } => {
-            // XXX currently, this error report goes nowhere except to the log.
-            // The Upstairs should track this for each Downstairs.
+            // The Upstairs should not consider a job completed until it has
+            // returned an Ok result, and should therefore log and eat all
+            // ErrorReport messages here. This will change in the future when
+            // the Upstairs tracks the number of errors per Downstairs, and acts
```

Comment:
Let's file an issue for this lest we forget it.

Reply:
I filed https://github.com/oxidecomputer/crucible/issues/857, which is a general version of the work required (as the previous error handling code did count the number of errors per Downstairs!).